### PR TITLE
Update plan activity to write show results to disk

### PR DESCRIPTION
**Pull request by samrabelachew**
The json show results are needed for the policy check step. Instead of rerunning the show step later on, let's write the original results from the plan step into a separate json file in the root directory.

### CODE REVIEW

File: server/neptune/workflows/activities/terraform.go [hunk 1]
```diff
@@ -253,20 +261,28 @@ func (t *terraformActivities) TerraformPlan(ctx context.Context, request Terrafo
...
+	summary, err := terraform.NewPlanSummaryFromJSON(showResults)
 	if err != nil {
 		activity.GetLogger(ctx).Error("error building plan summary", key.ErrKey, err)
 	}
 
+	// TODO: optimize by skipping write step when not in PR Mode
+	// write show results to disk
+	planJSONFile := filepath.Join(request.Path, PlanOutputJSONFile)
+	if err := t.FileWriter.Write(planJSONFile, showResults); err != nil {
+		return TerraformPlanResponse{}, errors.Wrap(err, "writing terraform show results to disk")
```
Comment (nishkrishnan) [inline server/neptune/workflows/activities/terraform.go:279]: Hmm, we're intentionally not failing when we can't run the results of show, so i think we should do the same here as well.  That way our post-merge workflow will always succeed even if this is an issue and then when we run our conftest activity we can just fail the step if the file doesn't exist.

Comment (samrabelachew) [reply server/neptune/workflows/activities/terraform.go:279]: The argument I was trying to make here is that the show step is now critical so we should fail immediately. To make it easier for post merge workflow, I can just support the TODO now and state that we only run this in PRMode

Comment (nishkrishnan) [reply server/neptune/workflows/activities/terraform.go:279]: It's critical for policy checking, not for planning so you should just fail the policy checking step if the file isn't there. and we can figure out from the logs why the show step failed.

Comment (nishkrishnan) [reply server/neptune/workflows/activities/terraform.go:279]: Either way something is going to fail, it's just a matter of where you're defining it that I have concerns with.

Comment (samrabelachew) [reply server/neptune/workflows/activities/terraform.go]: removed errored returns and kept logs

**Pull request merged**